src/transaction/transaction_manager.cpp

```diff
@@ -55,9 +56,44 @@ timestamp_t TransactionManager::ReadOnlyCommitCriticalSection(TransactionContext
   return commit_time;
 }
 
+bool TransactionManager::CheckConstraints(TransactionContext *txn,
+                                          std::vector<TransactionConstraint *> *txn_installed_constraints) {
+  auto iter = constraints_.Begin();
+  while (iter != constraints_.End()) {
```

Comment:
Yeah, it's kind of dumb. When you use the C++ iterator for loop it looks for a .begin() function. I get an error say no .begin() function found because in the wrapper for ConcurrentVector they use .Begin()

Reply:
Is there a reason to not use a C++ iterator-for loop here?